Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -101,48 +102,15 @@ class ListAttributeTargetObserver final : public IdTargetObserver {
 };
 #endif
 
-// FIXME: According to HTML4, the length attribute's value can be arbitrarily
-// large. However, due to https://bugs.webkit.org/show_bug.cgi?id=14536 things
-// get rather sluggish when a text field has a larger number of characters than
-// this, even when just clicking in the text field.
-const unsigned HTMLInputElement::maxEffectiveLength = 524288;
-const int defaultSize = 20;
-const int maxSavedResults = 256;
+static constexpr int maxSavedResults = 256;
 
 HTMLInputElement::HTMLInputElement(const QualifiedName& tagName, Document& document, HTMLFormElement* form, bool createdByParser)
     : HTMLTextFormControlElement(tagName, document, form)
-    , m_size(defaultSize)
-    , m_isChecked(false)
-    , m_dirtyCheckednessFlag(false)
-    , m_isIndeterminate(false)
-    , m_hasType(false)
-    , m_isActivatedSubmit(false)
-    , m_autocomplete(Uninitialized)
-    , m_isAutoFilled(false)
-    , m_isAutoFilledAndViewable(false)
-    , m_isAutoFilledAndObscured(false)
-    , m_autoFillButtonType(static_cast<uint8_t>(AutoFillButtonType::None))
-    , m_lastAutoFillButtonType(static_cast<uint8_t>(AutoFillButtonType::None))
-    , m_isAutoFillAvailable(false)
-#if ENABLE(DATALIST_ELEMENT)
-    , m_hasNonEmptyList(false)
-#endif
-    , m_stateRestored(false)
     , m_parsingInProgress(createdByParser)
-    , m_valueAttributeWasUpdatedAfterParsing(false)
-    , m_wasModifiedByUser(false)
-    , m_canReceiveDroppedFiles(false)
-#if ENABLE(TOUCH_EVENTS)
-    , m_hasTouchEventHandler(false)
-#endif
-    , m_isSpellcheckDisabledExceptTextReplacement(false)
-    , m_hasPendingUserAgentShadowTreeUpdate(false)
-{
     // m_inputType is lazily created when constructed by the parser to avoid constructing unnecessarily a text inputType,
     // just to destroy them when the |type| attribute gets set by the parser to something else than 'text'.
-    if (!createdByParser)
-        m_inputType = InputType::createText(*this);
-
+    , m_inputType(createdByParser ? nullptr : RefPtr<InputType> { InputType::createText(*this) })
```

Comment:
Maybe just `RefPtr` instead of `RefPtr<InputType>` and if that doesn’t work we should add a deduction guide so it does at some point.

Reply:
That's what I tried at first but it didn't work. I can look in deduction guides.